01-supp-addressingData.Rmd

```diff
@@ -0,0 +1,168 @@
+---
+layout: page
+title: Programming with R
+subtitle: Addressing data
+minutes: 20
+---
+
+
+```{r, include = FALSE}
+source('tools/chunk-options.R')
+```
+
+> ## Objectives {.objectives}
+>
+> Understand the 3 different ways R can address data inside a data frame.
+>
+> Combine different methods for addressing data with the assignment operator to update subsets of data
+
+R is a powerful language for data manipulation. There are 3 main ways for addressing data inside R objects.
+
+* By index (slicing)
+* By logical vector
+* By name (columns and lists only)
+
+Lets start by loading some sample data:
+
+```{r readData}
+dat<-read.csv(file='data/sample.csv',header=TRUE, stringsAsFactors=FALSE)
```

Comment:
We don't currently have an official stance on the use of `stringsAsFactors = FALSE`. A few comments:

These lessons are not only for instructors. The reason we spend so much time building the lessons into nicely formatted html is so that self-directed learners can go through the online lessons themselves.

In general, I think it is ok to sometimes say "don't worry about this for now." With workshops only lasting two days, we have to focus on getting across the big picture.

I think you should remove `header = TRUE` since that is the default for `read.csv`.